avail-intellij-plugin/src/main/kotlin/org/availlang/intellij/plugin/module/AvailProjectTemplate.kt

```diff
@@ -0,0 +1,244 @@
+  )
+
+  /**
+   * Creates a new Avail project.
+   *
+   * @param config The configuration for the project.
+   */
+  @JvmStatic
+  fun create (config: Config)
+  {
+    val projectFilePath = "${config.projectLocation}/${config.fileName}.json"
+    val configPath =
+      AvailEnvironment.projectConfigPath(config.fileName, config.projectLocation)
+    AvailProject.optionallyInitializeConfigDirectory(configPath)
+    val localSettings = LocalSettings.from(File(configPath))
+    AvailProjectV1(
+      config.fileName,
+      true,
+      AvailRepositories(rootNameInJar = null),
+      localSettings
+    ).apply {
+      File(config.projectLocation).mkdirs()
+      val rootsDir = config.rootsDir
+      val rootName = config.rootName
+      val rootLocationDir = rootsLocationDir(rootsDir, rootName)
+      File("${config.projectLocation}/$rootLocationDir").mkdirs()
+
+      val rootConfigDir = AvailEnvironment.projectRootConfigPath(
+        config.fileName,
+        rootName,
+        config.projectLocation)
+      val root = availProjectRoot(rootConfigDir, config, rootName, rootLocationDir)
+      addRoot(root)
+      optionallyInitializeConfigDirectory(rootConfigDir)
+      config.selectedLibrary?.let { lib ->
+        configureAvailStdLib(config, lib)
+      }
+
+      updateBackingProjectFile(projectFilePath, config)
+    }
+  }
+
+  /**
+   * Updates the backing project file with the current AvailProjectV1 instance.
+   *
+   * @param projectFilePath The file path of the backing project file.
+   * @param config The configuration settings.
+   */
+  private fun AvailProjectV1.updateBackingProjectFile(
+    projectFilePath: String,
+    config: Config
+  )
+  {
+    if (projectFilePath.isNotEmpty()) {
+      // Update the backing project file.
+      val writer = JSONWriter.newPrettyPrinterWriter()
+      writeTo(writer)
+      File(projectFilePath).writeText(writer.contents())
+      config.environmentSettings.add(this, projectFilePath)
+    }
+  }
+
+  /**
+   * Configures the Avail standard library for the given Avail project.
+   *
+   * @param config The configuration settings for the Avail project.
+   * @param lib The file representing the Avail standard library.
+   */
+  private fun AvailProjectV1.configureAvailStdLib(
+    config: Config,
+    lib: File
+  )
+  {
+    val libName = config.libraryName ?: AVAIL_STDLIB_ROOT_NAME
+    val libConfigDir = AvailEnvironment.projectRootConfigPath(
+      config.fileName, libName, config.projectLocation
+    )
+    optionallyInitializeConfigDirectory(libConfigDir)
+    val jar = AvailArtifactJar(lib.toURI())
+
+    val rootManifest = jar.manifest.roots[AVAIL_STDLIB_ROOT_NAME]
+    var sg = importStyles(config, jar)
+    var tg = importTemplates(config, jar)
+    val extensions = mutableListOf<String>()
+    var description = ""
+    rootManifest?.let {
+      sg = it.styles
+      tg = it.templates
+      extensions.addAll(it.availModuleExtensions)
+      description = it.description
+    }
```

Comment:
I know I wrote this code and you are cloning it, but feel free to do a cooler job than I did. I'm often pretty critical of the code I write as I like to keep getting better and writing cleaner code. So feel free when cloning something I did to improve upon it. I'm not sensitive about these things and I'm very comfortable in recognizing when code is done better than I originally did. This comment isn't about changing this code, just an acknowledgement that if there is a better, more Kotlin way to do something, than feel free to make those changes. 😀

With this for example, just a cursory sweep, I could've made this be a little neater (_probably could improve it some more than below_):

```kotlin
val description = rootManifest?.let {
	sg = it.styles
	tg = it.templates
	extensions.addAll(it.availModuleExtensions)
	it.description
} ?: ""
addRoot(
	AvailProjectRoot(
		libConfigDir,
		projLocation,
		libName,
		AvailLibraries(
		"org/availlang/${lib.name}",
			Scheme.JAR,
			AVAIL_STDLIB_ROOT_NAME),
		LocalSettings(
			AvailEnvironment.projectRootConfigPath(
				fileName,
				libName,
				projLocation)),
		sg,
		tg,
		extensions
	).apply {
		this.description = description
		saveLocalSettingsToDisk()
		saveTemplatesToDisk()
		saveStylesToDisk()
	})
```